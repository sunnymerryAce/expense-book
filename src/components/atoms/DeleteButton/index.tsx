import React from "react";
import IconButton from "@material-ui/core/IconButton";
import DeleteIcon from "@material-ui/icons/Delete";

interface Props {
  onClick: (props: any) => any;
  disabled?: boolean;
}

const DeleteButton: React.FC<Props> = (props) => {
  return (
    <IconButton
      aria-label="delete"
      onClick={props.onClick}
      disabled={props.disabled}
    >
      <DeleteIcon />
    </IconButton>
  );
};

export default DeleteButton;
