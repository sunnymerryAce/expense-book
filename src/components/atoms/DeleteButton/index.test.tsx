import React from "react";
import DeleteButton from "./";
import { shallow } from "enzyme";

describe("<DeleteButton />", () => {
  let container: any = null;
  const props = {
    handleClick: jest.fn(),
    disabled: true,
  };

  afterEach(() => {
    container.unmount();
    container = null;
  });

  test("should match the snapshot", () => {
    container = shallow(<DeleteButton {...props} />);
    expect(container.html()).toMatchSnapshot();
  });

  test("should have proper props", () => {
    container = shallow(<DeleteButton {...props} />);
    expect(container.find(".DeleteButton").props()).toMatchObject({
      onClick: props.handleClick,
      disabled: true,
    });
  });

  test("should have proper props(disable false)", () => {
    const props = {
      handleClick: jest.fn(),
      disabled: false,
    };
    container = shallow(<DeleteButton {...props} />);
    expect(container.find(".DeleteButton").props()).toMatchObject({
      onClick: props.handleClick,
      disabled: false,
    });
  });

  test("click event handler should be triggered", () => {
    container = shallow(<DeleteButton {...props} />);
    container.simulate("click");
    expect(props.handleClick).toBeCalled();
  });
});
