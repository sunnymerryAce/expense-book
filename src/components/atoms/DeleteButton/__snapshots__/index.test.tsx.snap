// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DeleteButton /> should match the snapshot 1`] = `"<button class=\\"MuiButtonBase-root MuiIconButton-root DeleteButton MuiIconButton-colorPrimary Mui-disabled Mui-disabled\\" tabindex=\\"-1\\" type=\\"button\\" disabled=\\"\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z\\"></path></svg></span></button>"`;
